test/integration/integration_test.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+require_relative '../test_helper'
+require_relative 'integration_helper'
+
+IntegrationTest.describe 'Integration testing with Rails app', '>= 2.5.3' do
+  include ShellHelper
+
+  RAILS_APP_PATH = File.expand_path('fakeapp', __dir__).freeze
+
+  before(:all) do
+    Dir.chdir(RAILS_APP_PATH) { sh('bundle install') }
+  end
+
+  around do |test|
+    Dir.chdir(RAILS_APP_PATH) { test.call }
+  end
+
+  it 'prints simple progress' do
+    output = sh("#{Bundler.root}/bin/bumbler")
+    output.must_include <<~TEXT
+      ( 0/10)  rails
+      ( 1/10)  sqlite3
+      ( 2/10)  puma
+      ( 3/10)  sass-rails
+      ( 4/10)  jbuilder
+      ( 5/10)  byebug
+      ( 6/10)  listen
+    TEXT
+    output.must_match(/\d+ of 10 gems required/)
```

Comment:
I didn't make this test tighter because the gem count stops incrementing properly when spring gets loaded. I think this has something special rails is doing with spring, but I do not have time to dig into this. My current fixes get this gem working for my large rails app, which was my main goal with this patch.